src/StructureMap.DynamicInterception/MethodInvocation.cs

```diff
@@ -39,6 +54,21 @@ public MethodInfo MethodInfo
             get { return _invocation.Method; }
         }
 
+        public MethodInfo MethodInfoInvocationTarget
+        {
+            get { return _invocation.MethodInvocationTarget; }
+        }
+
+        public object Proxy
```

Comment:
I don't really want to expose Castle proxy in this abstraction, since in theory it can be changed later to other implementation. Also I can't really think of use cases when it could be useful to have access to the proxy.